src/main/scala/is/hail/sparkextras/OrderedRDD2.scala

```diff
@@ -14,16 +14,22 @@ class OrderedDependency(left: OrderedRVD, right: OrderedRVD) extends NarrowDepen
 
 object OrderedDependency {
   def getDependencies(p1: OrderedRVDPartitioner, p2: OrderedRVDPartitioner)(partitionId: Int): Range = {
-    val lastPartition = if (partitionId == p1.rangeBounds.length)
-      p2.numPartitions - 1
-    else
-      p2.getPartitionPK(p1.rangeBounds(partitionId))
+    val partBounds = p1.rangeBounds(partitionId).asInstanceOf[Interval]
 
-    if (partitionId == 0)
-      0 to lastPartition
+    if (!p2.rangeTree.overlaps(p2.pkType.ordering, partBounds))
+      0 until 0
     else {
-      val startPartition = p2.getPartitionPK(p1.rangeBounds(partitionId - 1))
-      startPartition to lastPartition
+      val startPart = p2.rangeTree.queryValues(p2.pkType.ordering, partBounds.start)
+      val endPart = p2.rangeTree.queryValues(p2.pkType.ordering, partBounds.end)
+      val start = startPart match {
+        case Array() => p2.numPartitions - 1
```

Comment:
Ah, don't you want 0 for the start?  And maybe throw an assert in here that partBounds.start < partition min?